packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -168,6 +189,9 @@ Future<void> _runCmakeGeneration({
         buildDir.path,
         '-G',
         generator,
+        '-A',
+        arch,
+        '-DFLUTTER_TARGET_PLATFORM=$flutterTargetPlatform',
```

Comment:
Can we move this to the generated configs in `_writeGeneratedFlutterConfig`?

https://github.com/flutter/flutter/blob/b7acafa0bbf6a23943c447eb98a6b21b1adcbe93/packages/flutter_tools/lib/src/cmake.dart#L101-L109

Reply:
Linux build adds it directly to the cmake command, and IMHO, it's more explicit that putting it in another function.
FLUTTER_VERSION_* stuff is the same for all architecture, thus it makes sense to factorize it in a single place.

What do you think?

Reply:
Hm I agree we should align with Linux. If we move `FLUTTER_TARGET_PLATFORM` to the generated CMake config file, we would need to also update Linux. That seems out-of-scope for this PR, and we can follow-up with this change later if needed.

That said, I'm curious as to why Linux chose this approach. The generated cmake config file makes it easier to use the CMake project without the Flutter tool. Regardless, I'll close this off!

Reply:
Thanks for accepting this.

Reply:
What happens if someone directly invokes a build via cmake without providing `-DFLUTTER_TARGET_PLATFORM`?